src/fsharp/FSharp.Core/async.fsi

```diff
@@ -299,8 +532,49 @@ namespace Microsoft.FSharp.Control
         /// <returns>A computation that returns the first succeeding computation.</returns>
         ///
         /// <category index="1">Composing Async Computations</category>
-        /// 
-        /// <example-tbd></example-tbd>
+        ///
+        /// <example id="choice-example-1">
+        /// <code lang="fsharp">
+        /// printfn "Starting"
+        /// [ 2; 3; 5; 7 ]
+        /// |> List.map
+        ///     (fun i ->
+        ///         async {
+        ///             do! Async.Sleep(System.Random().Next(1000, 2000))
```

Comment:
Define the random number generator outside, e.g. `let rnd = System.Random()`

Reply:
[System.Random objects are not threadsafe](https://docs.microsoft.com/en-us/dotnet/api/system.random?view=net-6.0#ThreadSafety), which is why I create a new one for each asynchronous computation.

We could define one and use a locking mechanism, but for the given example there seems to be no purpose, and while there's every reason to want a `System.Random` with a specific seed inside an order-deterministic series of computations for reproducibility, it makes little sense to share a `System.Random` between order-nondeterministic computations.

Reply:
With .net framework the Random instances are likely to be initialized with identical seeds. Even with .net core I don't think we should encourage using a Random instances once for one number.

https://docs.microsoft.com/en-us/dotnet/api/system.random?view=net-6.0#avoiding-multiple-instantiations

Using a lock would probably be preferred.

See the F# examples [here](https://docs.microsoft.com/en-us/dotnet/api/system.random?view=net-6.0#the-systemrandom-class-and-thread-safety)